website/docs/migration/from_state_notifier.mdx

```diff
@@ -0,0 +1,386 @@
+Furthermore, unlike every other provider, the `dispose` callback is to be defined
+in the class, separately.  
+Finally, it is still possible to write `ref.onDispose` in its _provider_, showing once again how
+sparse the logic can be with this API; potentially, the developer might have to look into six (6!)
+different places to understand this Notifier behavior!
+
+Again, this ambiguity is solved with `Riverpod 2.0`.
+Check the above example, migrated to its `Notifier` equivalent.
+
+<AutoSnippet language="dart" {...oldLifecycles}></AutoSnippet>
+
+This simplifies the API and, hopefully, your DX, for two reasons:
+
+1. There is only _one_ lifecycle to take care of, and it's managed via `ref.onDispose`
+   (and `ref.onCancel`, etc.), just like you would do with any other provider
+2. There is only _one_ place to look at to understand lifecycle side-effects, its `build` method
+3. Also, the above `MyNotifier`, will _not_ dispose when `durationProvider` updates:
+   only its _internal state_ will
+
+### `.mounted` is no more
+
+**TODO: talk about how there is no `.mounted` property and how to transition from it**
+
+## Other migrations
+
+Let's explore the less-impactful differences between `StateNotifier` and `Notifier` (or `AsyncNotifier`)
+
+### From `.addListener` and `.stream`
+
+`StateNotifier`'s `.addListener` and `.stream` can be used to listen for state changes.
+These two APIs are now to be considered outdated.
+
+This is intentional due to the desire to reach full API uniformity with `Notifier`, `AsyncNotifier` and other providers.  
+Indeed, using a `Notifier` or an `AsyncNotifier` shouldn't be any different from any other provider.
+
+Therefore this:
+```dart
+class MyNotifier extends StateNotifier<int> {
+  MyNotifier() : super(0);
+}
+
+final myNotifierProvider = StateNotifierProvider<MyNotifier, int>((ref) {
+  final notifier = MyNotifier();
+
+  final cleanup = notifier.addListener((state) => debugPrint('$state'));
+  ref.onDispose(cleanup);
+
+  // Or, equivalently:
+  // final listener = notifier.stream.listen((event) => debugPrint('$event'));
+  // ref.onDispose(listener.cancel);
+
+  return notifier;
+});
+```
+Becomes this:
+<AutoSnippet language="dart" {...addListener}></AutoSnippet>
+
+In a nutshell: if you want to listen to a `Notifier`/`AsyncNotifer`, just use `ref.listen`,
+[as explained here](/docs/concepts/reading#using-reflisten-to-react-to-a-provider-change).
+
+### From `.debugState`
+
+`StateNotifier` exposed `.debugState`: this property had some use within `state_notifier`,
+as it enabled accessing state from outside the class when in development mode, for testing purposes.
+
+Again, for uniformity reasons, such property is not available in the new APIs.  
+If you want to interact with `Notifier`/`AsyncNotifier` in a testing environment, you should do
+it just like with any other provider: by reading them directly (as opposed of directly instantiating
+a `Notifier` by hand).
```

Comment:
Not sure what you mean.

`Notifier.state` is accessible by tests if they wanted to. It's `@visibleForTesting`.

Reply:
I'm trying to write down your bullet points in #2630.

There, I read:
> StateNotifier.debugState no longer exists.
Tests which want to interact with Notifiers should always do so by reading the provider instead of directly instantiating the Notifier by hand.
You can do providerContainer.read(provider) to get the current state of a provider inside tests.

So, what do I do with this? I can delete the whole paragraph if you want me to.

Reply:
Ah right. Those bullet points aren't quite organized. It's mix and matching a few things

I'd split it in two. There are two things going on:
- Notifiers shouldn't be instantiated by hand. They should be interacted with by using the provider
  Failing to do so will break the notifier due to ref & family args not being initialized
- StateNotifier.debugState doesn't exist. If you have a notifier, you can access `.state` inside tests freely.
  If you don't have the Notifier instance, read the provider to obtain one. Cf above